### PR TITLE
Add test stub hygiene guard

### PR DESCRIPTION
**Pull request by eXPerience83**
### Closed in favor of a smaller replacement PR

This PR became too broad while trying to cover many AST edge cases. The original intent was useful, but the implementation started drifting toward a custom linter.

Closing this in favor of a smaller, focused hygiene guard that only blocks the concrete import-time stub patterns this cleanup originally wanted to prevent.

### CODE REVIEW

File: tests/test_stub_hygiene.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+STUB_INSTALL_HELPERS = {
+    "force_module",
+    "stub_aiohttp_module",
+    "stub_config_entry_class",
+    "stub_custom_components_packages",
+    "stub_exceptions",
+    "stub_homeassistant_package",
+    "stub_update_coordinator_module",
+}
```
[the remark this answers is not in the document]
Comment (eXPerience83) [reply tests/test_stub_hygiene.py:22]: @gemini-code-assist please check again.
Addressed in the latest commit.

`clear_integration_modules` has been added to `STUB_INSTALL_HELPERS`, so top-level calls to that helper are now detected by the hygiene guard.

File: tests/test_stub_hygiene.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    def visit_AnnAssign(self, node: ast.AnnAssign) -> None:
+        if self.scope_depth == 0:
+            self._check_assignment_target(node.target)
+        self.generic_visit(node)
+
+    def visit_AugAssign(self, node: ast.AugAssign) -> None:
+        if self.scope_depth == 0:
+            self._check_assignment_target(node.target)
+        self.generic_visit(node)
```
[the remark this answers is not in the document]
Comment (eXPerience83) [reply tests/test_stub_hygiene.py:87]: @gemini-code-assist please check again.
Addressed in the latest commit.

The visitor now implements `visit_Delete()` and reports top-level `del sys.modules[...]` usage as a `top-level sys.modules deletion` violation.

File: tests/test_stub_hygiene.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    def _visit_nested_scope(
+        self, node: ast.FunctionDef | ast.AsyncFunctionDef | ast.ClassDef
+    ) -> None:
+        self.scope_depth += 1
+        self.generic_visit(node)
+        self.scope_depth -= 1
+
+    def _check_assignment_target(self, target: ast.expr) -> None:
+        if isinstance(target, ast.Subscript) and self._is_sys_modules(target.value):
+            self.violations.append((target.lineno, "top-level sys.modules assignment"))
```
[the remark this answers is not in the document]
Comment (eXPerience83) [reply tests/test_stub_hygiene.py:143]: @gemini-code-assist please check again.
Addressed in the latest commit.

The assignment checker now detects direct top-level reassignment of `sys.modules` and reports it as a `top-level sys.modules reassignment` violation.
